### DISCUSSION TITLE
Passing recipe parameters to dependency recipes

### DISCUSSION THREAD

**Discussion by nicholaschiasson** [casey/just#1302] [Q&A] [answered]
I want to do something like this but am not sure if it is possible. From what I have read in the documentation, it seems like an unsupported case.

```
a *FLAGS:
  ...

# b depends on a and passes its input FLAGS to a
b *FLAGS: a
  ...
```

Is it possible to accomplish something like this? Where a recipe passes its parameters to its dependency recipe?

**Comment by casey** [comment, marked as answer]
Yup, this is possible:

```
a *FLAGS:
  ...

# b depends on a and passes its input FLAGS to a
b *FLAGS: (a FLAGS)
  ...
```

**Reply by zpah** [reply]
I have tried the suggested method in my justfile, but i found _weird_ _idiosyncrasy_! :)

In this test case receipt "good" is successful, but simple wrapper receipt "expected_good" failed:
```
set positional-arguments

a *FLAGS:
  #!/usr/bin/env bash
  set -eux
  typeset -a argv=("$@")
  declare -p argv
  [[ ${1} == 'f1' ]]
  [[ ${2} == 'f2' ]]
  [[ ${3} == 'f3 "x y z"' ]]

b *FLAGS: (a FLAGS)

good: (a 'f1' 'f2' 'f3 "x y z"')

expected_good: (b 'f1' 'f2' 'f3 "x y z"')
```
Is it a _just_ issue or my mistake?
If it is a _just_ issue, what is the workaround?

**Reply by nicholaschiasson** [reply]
Interesting. I was playing with your example trying to make it work and couldn't. So something changes possibly about escaped strings when you pass args multiple recipes deep... 🤔

**Reply by laniakea64** [reply]
@zpah try this `b`? -
```
b *FLAGS:
  just -f {{quote(justfile())}} a "$@"
```

**Reply by nicholaschiasson** [reply]
May work but aside from being a bit more clunky, this isn't the same as depending on the other recipe. Nothing is stopping other recipes from running `a` redundantly, which matters for some workflows.

**Reply by zpah** [reply]
Thank you for response!
I agree with @nicholaschiasson, this is too fragile solution leading to errors.
It breaks, for example, when needs pass extra variable from command line:
```
$ just --set var1 '2' b
```
In nested _just_ invocation, the new var1 value will not be visible.
Fixing will require further obfuscation of the code.

Issue resolved by extending the justfile syntax with the _List_ type.
Receipt parameters are need considered as _List_ in this case.
